substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -287,7 +287,10 @@ impl<T: Config> Token<T> for RuntimeCosts {
 			WeightToFee => T::WeightInfo::seal_weight_to_fee(),
 			Terminate(locked_dependencies) => T::WeightInfo::seal_terminate(locked_dependencies),
 			Random => T::WeightInfo::seal_random(),
-			DepositEvent { num_topic, len } => T::WeightInfo::seal_deposit_event(num_topic, len),
+			// Given a 2-second block time and hardcoding a `ref_time` of 60,000 picoseconds per
+			// byte, the max allocation size is 32MB per block.
+			DepositEvent { num_topic, len } => T::WeightInfo::seal_deposit_event(num_topic, len)
+				.saturating_add(Weight::from_parts(60_000u64.saturating_mul(len.into()), 0)),
```

Comment:
Thoughts about moving that in a pallet::constant or maybe as a new field in Limits (src/schedule) so that chain with different constraints can adapt it

Also I would add in the comment the full formula that led you to the 32MB result so that other chain can adapt that number to their parameters

Reply:
Formula with comment is here: https://github.com/paritytech/polkadot-sdk/blob/5cf2ef25b50de16d815e325fb8c63c62fd40274c/substrate/frame/contracts/src/lib.rs#L700